lib/Profile/TalkAction.php

```diff
@@ -0,0 +1,112 @@
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+namespace OCA\Talk\Profile;
+
+use OCA\Talk\AppInfo\Application;
+use OCP\IURLGenerator;
+use OCP\IUser;
+use OCP\IUserManager;
+use OCP\IUserSession;
+use OCP\L10N\IFactory;
+use OCP\Profile\IAction;
+
+class TalkAction implements IAction {
+
+	/** @var string */
+	private $userId;
+
+	/** @var IFactory */
+	private $l10nFactory;
+
+	/** @var IUserManager */
+	private $userManager;
+
+	/** @var IUserSession */
+	private $userSession;
+
+	/** @var IUrlGenerator */
+	private $urlGenerator;
+
+	public function __construct(
+		IFactory $l10nFactory,
+		IUserManager $userManager,
+		IUserSession $userSession,
+		IURLGenerator $urlGenerator
+	) {
+		$this->l10nFactory = $l10nFactory;
+		$this->userManager = $userManager;
+		$this->userSession = $userSession;
+		$this->urlGenerator = $urlGenerator;
+	}
+
+	public function preload(IUser $user): void {
+		$this->userId = $user->getUID();
+	}
+
+	public function getAppId(): string {
+		return Application::APP_ID;
+	}
+
+	public function getId(): string {
+		return 'talk';
+	}
+
+	public function getTitle(): string {
+		$visitingUser = $this->userSession->getUser();
+		$targetUser = $this->userManager->get($this->userId);
+		if ($visitingUser === $targetUser) {
+			return $this->l10nFactory->get(Application::APP_ID)->t('Open Talk');
+		}
+		return $this->l10nFactory->get(Application::APP_ID)->t('Talk to %s', [$targetUser->getDisplayName()]);
+	}
+
+	public function getLabel(): string {
+		$visitingUser = $this->userSession->getUser();
+		$targetUser = $this->userManager->get($this->userId);
+		if ($visitingUser === $targetUser) {
+			return $this->l10nFactory->get(Application::APP_ID)->t('Open Talk');
+		}
+		return $this->l10nFactory->get(Application::APP_ID)->t('Talk');
+	}
+
+	public function getPriority(): int {
+		return 10;
+	}
+
+	public function getIcon(): string {
+		return $this->urlGenerator->getAbsoluteURL($this->urlGenerator->imagePath('spreed', 'app-dark.svg'));
+	}
+
+	public function getTarget(): string {
+		$visitingUser = $this->userSession->getUser();
+		$targetUser = $this->userManager->get($this->userId);
+		if ($visitingUser === $targetUser) {
+			return $this->urlGenerator->linkToRouteAbsolute('spreed.Page.index');
+		}
+		return $this->urlGenerator->linkToRouteAbsolute('spreed.Page.index') . '?callUser=' . $this->userId;
```

Comment:
`$visitingUser` is null for guests and that will make the action redirect to a broken page.
If the actions are also used for guests it shoulduse:
```suggestion
		if (!$visitingUser || $visitingUser === $targetUser) {
			return $this->urlGenerator->linkToRouteAbsolute('spreed.Page.index');
		}
		return $this->urlGenerator->linkToRouteAbsolute('spreed.Page.index') . '?callUser=' . $this->userId;
```

Reply:
Also similar to the call action in the contacts menu we need to check if the talk app is enabled for the target users:
https://github.com/nextcloud/spreed/blob/84d42604e60aa1c31ecf698dcc069f36c494148a/lib/ContactsMenu/Providers/CallProvider.php#L81-L84